### PR TITLE
Runtime: Support for time_range functions

### PR DESCRIPTION
**Pull request by k-anshul**
Added two function:
1. `time_range_start(duration, [unit, col])` and `time_range_end(duration, [unit, col])`
2. `time_range` will be added in later phases.

### CODE REVIEW

File: runtime/pkg/duration/duration.go [hunk 1]
```diff
@@ -32,11 +32,11 @@ var (
 	}
 	daxOffsetNotations = map[string]StandardDuration{
 		"PP": {},
```
Comment (k-anshul) [inline runtime/pkg/duration/duration.go:34]: What should we do wrt `PP`? Add an explicit check that it is not allowed in time_range functions

**Comment by k-anshul** [comment]
No clue why e2e tests are failing on this PR. The changes are only in `runtime/pkg` and pretty much self contained.

File: runtime/resolvers/metrics_sql.go [hunk 2]
```diff
@@ -47,7 +47,12 @@ func newMetricsSQL(ctx context.Context, opts *runtime.ResolverOptions) (runtime.
 		return nil, err
 	}
 
-	compiler := metricssqlparser.New(ctrl, opts.InstanceID, opts.UserAttributes)
+	sqlArgs := &sqlArgs{}
+	if err := mapstructure.Decode(opts.Args, sqlArgs); err != nil {
+		return nil, err
+	}
+
+	compiler := metricssqlparser.New(ctrl, opts.InstanceID, opts.UserAttributes, sqlArgs.Priority)
```
Comment (begelundmuller) [inline runtime/resolvers/metrics_sql.go:55]: A little confusing to overload `sqlArgs` – maybe add an explicit `metricsSQLArgs` type?

**Review by begelundmuller — APPROVED** [review]

File: runtime/resolvers/metrics_sql.go (hunk 2 above)
Comment (k-anshul) [reply runtime/resolvers/metrics_sql.go]: Sure. Fixed.

**Comment by AndrewRTsao** [comment]
@k-anshul Should we add a snippet to our docs to call out support for time_range functions?

**Pull request merged**